### ISSUE TITLE
Calling set_follow_target_offset breaks Phantom Camera initial properties

### ISSUE THREAD

**Issue by madeleineostoja** [ramokz/phantom-camera#159]
### Issue description

Disclaimer that I'm a software engineer who's pretty new to Godot & gamedev in general so this could be user error, but I'm trying to create a simple "look ahead" type behaviour with a `PhantomCamera2D`, and calling `set_follow_target_offset` from a signal that's called when the player's direction changes seems to reset all initial properties on my phantom camera (eg: damping, framed follow dead zone, etc). 

### Steps to reproduce

1. Setup a PhantomCamera2D with Framed follow mode on a player object and camera damping
2. Call `set_follow_target_offset` 
3. See initial camera behaviour be broken

Code:
```
@export var camera_offset = 100
@onready var camera: PhantomCamera2D = $PhantomCamera

func _on_player_direction_changed(direction: Vector2) -> void:
  camera.set_follow_target_offset(direction * camera_offset)
```

Behaviour:

https://github.com/ramokz/phantom-camera/assets/6374876/3149a47c-a5b4-4981-a2b1-c732a6737337

**Comment by ramokz** [comment]
That sounds odd! Nothing should be resetting any of the initial properties by changing the `follow_target_offset`.

Does this also happen when using the `Simple Follow` mode, or if you greatly reduce the `camera_offset` amount?

My best guess would be that it's related to the `Framed Follow` logic specifically. Where the target position that is being tracked, and in your case offset, exceeds the bounds of the `Dead Zones`, which can cause its follow logic to effectively break. This is not an uncommon issue, and is _likely_ what you're seeing, though it's hard to say exactly without debugging it. The `Framed Follow` does need a proper makeover, as it has been running into various issues, among this (assuming that is the cause behind it).

Funnily enough, the day after this was posted, a PR was set up to support `Look Ahead` (#160), so there is at least a plan for getting built-in support for it. Though I very much doubt it would be a silver bullet for `Framed Follow` if, again, that is indeed the root cause here.

**Comment by madeleineostoja** [comment]
> My best guess would be that it's related to the Framed Follow logic specifically. Where the target position that is being tracked, and in your case offset, exceeds the bounds of the Dead Zones, which can cause its follow logic to effectively break. 

Yep! Looks like this is the issue. Both reducing `camera_offset` to a small value (which doesn't move the camera offset at all, I assume because of the same edge case?) and changing to simple follow seems to fix the issue

For now the framed follow is more important for my use case, but I'll keep an eye on the PR for look ahead and see if it gets support for framed follow

**Comment by ramokz** [comment]
I'll try and shift priorities around and do a proper refactor for `Framed Follow` in the not too distant future.
Been seeing more issues popping up for it lately.

**Comment by madeleineostoja** [comment]
Thank you! I’m not deep enough in Godot yet to offer help on a PR but happy to test and debug if needed

**Comment by ramokz** [comment]
Would be great to have additional eyes on it to make sure it works correctly!

Will make a draft PR at some point and link back to it here, and in the other issues, once I get around to it.

**Comment by Mustfa-IT** [comment]
I tried reproducing this issue on the current `main` branch, but I wasn’t able to observe the problem.

This is my current scene setup:

<img width="263" height="419" alt="Image" src="https://github.com/user-attachments/assets/fa5011cc-f2a0-462f-aed8-9d1baf1cd251" />

**Player script:**

```gdscript
extends CharacterBody2D

@export var camera_offset = 100
@export var camera: PhantomCamera2D

func _on_player_direction_changed(direction: Vector2) -> void:
	camera.set_follow_offset(direction * camera_offset)

func _physics_process(_delta: float) -> void:
	var direction := Input.get_axis("ui_up", "ui_down")
	if direction:
		_on_player_direction_changed(Vector2(0, direction))
		velocity.y = direction * 300
	else:
		velocity.y = move_toward(velocity.y, 0, 300)

	move_and_slide()
```

Here’s a short video showing the current behavior on my end:

https://github.com/user-attachments/assets/8500deb4-60ac-44b7-829c-69493a63fbdf

For reference, I’ve attached a minimal repro project. The scene used in the video is `node_2d.tscn` located at the project root:
[new-game-project.zip](https://github.com/user-attachments/files/24367133/new-game-project.zip)

Edit note: the `set_follow_target_offset` is not present anymore in the source code so i used `set_follow_offset`

**Comment by Mustfa-IT** [comment]
@madeleineostoja could you confirm whether this issue is still present on the current main branch? If so, I’d appreciate more detailed steps to reproduce it or a project with the issue presented, as I’m planning to work on this issue.

**Comment by Mustfa-IT** [comment]
@ramokz can you confirm wither or not this issue is still present?

**Comment by ramokz** [comment]
Thanks for looking into this and sharing the project above, @Mustfa-IT!
I can't repro the issue either, even when pushing the offset to an even higher value. It has likely been resolved as a side effect of various systemic changes that have occurred over time.

Will close the issue, but obviously feel free to reopen it if something isn't working still.

https://github.com/user-attachments/assets/a4673045-5852-4b95-90e7-8112e1be7e6c
